### PR TITLE
(FACT-2006) Add rake task for generating fact templates.

### PR DESCRIPTION
**Pull request by BogdanIrimie**

### CODE REVIEW

File: tasks/fact_generator/fact_creator.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      fact_name = fact['fact_name']
+
+      path = create_directory_path(operating_system, fact_name)
+      create_directory_structure(path)
+      create_fact_file(path, fact_name, operating_system)
+    end
+  end
+
+  def create_fact(operating_system, fact_name)
+    puts fact_name
```
Comment (ciprianbadescu) [inline tasks/fact_generator/fact_creator.rb]: are we allowed to use **puts**?
I would expect to use logging instead.

File: tasks/fact_generator/fact_creator.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+class FactCreator
+  def create_facts
+    facts_json_string = File.read(File.join(ROOT_DIR, 'tasks', 'fact_generator', 'facts.json'))
+    facts = JSON.parse(facts_json_string)
+
+    facts.each do |fact|
+      operating_system = fact['os']
+      fact_name = fact['fact_name']
+
+      path = create_directory_path(operating_system, fact_name)
```
Comment (ciprianbadescu) [inline tasks/fact_generator/fact_creator.rb]: should here be called create_fact instead duplicate code?

File: tasks/fact_generator/fact_creator.rb (hunk 1 above)
Comment (IrimieBogdan) [reply tasks/fact_generator/fact_creator.rb]: Replaced with logger.

File: tasks/create_fact.rake [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+desc 'Create a fact template for the specified arguments'
+task :create_fact, [:os, :fact_name] do |_, args|
```
Comment (gimmyxd) [inline tasks/create_fact.rake]: This should fail gracefully if os/fact_name are not passed

File: tasks/fact_generator/fact_creator.rb (hunk 2 above)
Comment (IrimieBogdan) [reply tasks/fact_generator/fact_creator.rb]: Replaced with call to create_fact

File: tasks/create_fact.rake (hunk 3 above)
Comment (IrimieBogdan) [reply tasks/create_fact.rake]: Added graceful fail when no arguments are provided.

**Review by sebastian-miclea — APPROVED** [review]

**Pull request merged**